staging/src/k8s.io/apiserver/pkg/storage/etcd3/store.go

```diff
@@ -274,7 +274,20 @@ func (s *store) GuaranteedUpdate(
 	transformContext := authenticatedDataString(key)
 	for {
 		if err := preconditions.Check(key, origState.obj); err != nil {
-			return err
+			// If our data is already up to date, return the error
```

Comment:
Can the new code be put in a helper func so that code is reused between this and the s.updateState case below ?

Reply:
if it doesn't make it harder to follow, that's possible. I wanted to keep this very very simple for backports.